### PR TITLE
[4.0] Fix some typehints in finder plugins to namespaced versions

### PR DESCRIPTION
**Pull request by wilsonge**
### Summary of Changes
Change typehints + unnamepsaced class usage to namespaced class usage

### Testing Instructions
Code reiew

### CODE REVIEW

File: plugins/finder/categories/categories.php [hunk 1]
```diff
@@ -10,18 +10,22 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Table\Table;
+use Joomla\Component\Finder\Administrator\Controller\IndexController;
```
Comment (SharkyKZ) [inline plugins/finder/categories/categories.php]:
```suggestion
```
This doesn't seem to be used in the file.

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 75d3c08093ba8f9de146ac40be431a55f1f4ec85<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26384">issues.joomla.org/tracker/joomla-cms/26384</a>.</sub>

File: administrator/components/com_finder/Indexer/Adapter.php [hunk 2]
```diff
@@ -383,7 +384,7 @@ protected function remove($id)
 	/**
 	 * Method to update index data on category access level changes
 	 *
-	 * @param   JTable  $row  A JTable object
+	 * @param   Table  $row  A JTable object
```
Comment (Quy) [inline administrator/components/com_finder/Indexer/Adapter.php]: Update description?

**Comment by richard67** [comment]
LGTM. But shouldn't the descriptions not be changed, too, like Quy mentioned? I.e.
`@param   Table  $row  A JTable object` => `@param   Table  $row  A Table object`
and so on?

**Comment by roland-d** [comment]
@wilsonge Please fix the conflicts.

**Comment by Hackwar** [comment]
Sorry, my merge didn't turn out well. The use statements at the top of the finder plugins were doubled. I created a replacement PR with #33920

**Comment by richard67** [comment]
@Hackwar @wilsonge So this PR here can be closed?

**Comment by Hackwar** [comment]
yes

**Comment by richard67** [comment]
Closing in favour of #33920 .

**Pull request closed without merging**